scripts/qemu-check.sh

```diff
@@ -7,110 +7,133 @@ READY_MSG="6c 00 00 00 00 00 00 70"
 
 rm -f dump-*.txt
 
-if [ "$#" -eq 0 ]
-then
-	PLATFORMS=${SUPPORTED_PLATFORMS[@]}
+print_usage()
+{
+	cat <<EOF
+usage: qemu-check.sh platform(s)
+	Supported platforms are ${SUPPORTED_PLATFORMS[*]}
+EOF
+}
+
+while getopts "" OPTION; do
+	case "$OPTION" in
+		*) print_usage; exit 1 ;;
+	esac
+done
+shift $((OPTIND-1))
+
+PLATFORMS=()
+if [ "$#" -eq 0 ]; then
+	PLATFORMS=("${SUPPORTED_PLATFORMS[@]}")
 else
-	for args in $@
-	do
-		for i in ${SUPPORTED_PLATFORMS[@]}
-		do
-			if [ $i == $args ]
-			then
-				PLATFORMS+=$i" "
+	for arg in "$@"; do
+		platform=none
+		for i in "${SUPPORTED_PLATFORMS[@]}"; do
+			if [ x"$i" = x"$arg" ]; then
+				PLATFORMS=("${PLATFORMS[@]}" "$i")
+				platform=$i
+				shift
+				break
 			fi
 		done
+		if [ "$platform" = "none" ]; then
+			echo "Error: Unknown platform specified: $arg"
+			echo "Supported platforms: ${SUPPORTED_PLATFORMS[*]}"
+			exit 1
+		fi
 	done
 fi
 
 # check target platform(s) have been passed in
 if [ ${#PLATFORMS[@]} -eq 0 ];
 then
-	echo "Error: No platforms specified. Supported are: ${SUPPORTED_PLATFORMS[@]}"
+	echo "Error: No platforms specified. Supported are: "\
```

Comment:
Did you try to trigger and test this (new) error message? I don't think you can. If this error cannot be triggered then it's pointless, remove it.

Reply:
Right, because if nothing is specified, it uses all platforms..